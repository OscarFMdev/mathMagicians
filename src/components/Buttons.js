import PropTypes from 'prop-types';

const Buttons = ({ myFunction }) => (
  <>
    <button onClick={myFunction} type="button" className="blue-button">AC</button>
    <button onClick={myFunction} type="button" className="blue-button">+/-</button>
    <button onClick={myFunction} type="button" className="blue-button">%</button>
    <button onClick={myFunction} type="button" className="orange-button">รท</button>
    <button onClick={myFunction} type="button" className="blue-button">7</button>
    <button onClick={myFunction} type="button" className="blue-button">8</button>
    <button onClick={myFunction} type="button" className="blue-button">9</button>
    <button onClick={myFunction} type="button" className="orange-button">x</button>
    <button onClick={myFunction} type="button" className="blue-button">4</button>
    <button onClick={myFunction} type="button" className="blue-button">5</button>
    <button onClick={myFunction} type="button" className="blue-button">6</button>
    <button onClick={myFunction} type="button" className="orange-button">-</button>
    <button onClick={myFunction} type="button" className="blue-button">1</button>
    <button onClick={myFunction} type="button" className="blue-button">2</button>
    <button onClick={myFunction} type="button" className="blue-button">3</button>
    <button onClick={myFunction} type="button" className="orange-button">+</button>
    <button onClick={myFunction} type="button" className="blue-button large">0</button>
    <button onClick={myFunction} type="button" className="blue-button">.</button>
    <button onClick={myFunction} type="button" className="orange-button">=</button>
  </>
);

Buttons.propTypes = {
  myFunction: PropTypes.func.isRequired,
};

export default Buttons;
