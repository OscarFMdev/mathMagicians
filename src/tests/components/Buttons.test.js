import { fireEvent, render, screen } from '@testing-library/react';
import Buttons from '../../components/Buttons';

describe('Buttons', () => {
  const handleClick = jest.fn();
  test('should render', () => {
    const { container } = render(<Buttons handleClick={handleClick} />);
    expect(container).toMatchSnapshot();
  });
  test('should contain "=" sign', () => {
    render(<Buttons handleClick={handleClick} />);
    const result = screen.getByText('=');
    expect(result).toBeDefined();
  });
  test('should call the handleClick function when clicked', () => {
    const { container } = render(<Buttons handleClick={handleClick} />);
    const buttons = container.querySelectorAll('button');
    const button = buttons[9];

    fireEvent.click(button);

    expect(handleClick).toBeCalledTimes(1);
  });
});
