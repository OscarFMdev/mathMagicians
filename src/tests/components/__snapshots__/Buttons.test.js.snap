// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should render 1`] = `
<div>
  <button
    class="blue-button"
    type="button"
  >
    AC
  </button>
  <button
    class="blue-button"
    type="button"
  >
    +/-
  </button>
  <button
    class="blue-button"
    type="button"
  >
    %
  </button>
  <button
    class="orange-button"
    type="button"
  >
    รท
  </button>
  <button
    class="blue-button"
    type="button"
  >
    7
  </button>
  <button
    class="blue-button"
    type="button"
  >
    8
  </button>
  <button
    class="blue-button"
    type="button"
  >
    9
  </button>
  <button
    class="orange-button"
    type="button"
  >
    x
  </button>
  <button
    class="blue-button"
    type="button"
  >
    4
  </button>
  <button
    class="blue-button"
    type="button"
  >
    5
  </button>
  <button
    class="blue-button"
    type="button"
  >
    6
  </button>
  <button
    class="orange-button"
    type="button"
  >
    -
  </button>
  <button
    class="blue-button"
    type="button"
  >
    1
  </button>
  <button
    class="blue-button"
    type="button"
  >
    2
  </button>
  <button
    class="blue-button"
    type="button"
  >
    3
  </button>
  <button
    class="orange-button"
    type="button"
  >
    +
  </button>
  <button
    class="blue-button large"
    type="button"
  >
    0
  </button>
  <button
    class="blue-button"
    type="button"
  >
    .
  </button>
  <button
    class="orange-button"
    type="button"
  >
    =
  </button>
</div>
`;
